Source/Fuse.Elements/Resources/Exif/Exif.uno

```diff
@@ -0,0 +1,190 @@
+using Uno;
+using Uno.IO;
+using Uno.Compiler.ExportTargetInterop;
+
+namespace Fuse.Resources.Exif
+{
+	[Flags]
+	public enum ImageOrientation
+	{
+		Identity = 0,
+		Rotate90 = 1,
+		Rotate180 = 2,
+		Rotate270 = 3,
+		FlipVertical = 1 << 2,
+	}
+
+	public struct ExifData
+	{
+		public readonly ImageOrientation Orientation;
+
+		public static ExifData FromByteArray(byte[] buffer)
+		{
+			if defined(Android)
+				return ExifAndroidImpl.FromByteArray(buffer);
+			if defined(iOS)
+				return ExifIOSImpl.FromByteArray(buffer);
+			if defined(DotNet)
+				return ExifDotNetImpl.FromByteArray(buffer);
+			else
+				throw new Exception("Exif is not supported by this platform");
+		}
+
+		internal ExifData(int orientation)
+		{
+			//This is based on the vertical flip being applied before the rotation
+			switch (orientation)
+			{
+				case 0: Orientation = ImageOrientation.Identity; break; //Exif orientation is undefined
+				case 1: Orientation = ImageOrientation.Identity; break;
+				case 2: Orientation = ImageOrientation.FlipVertical | ImageOrientation.Rotate180; break;
+				case 3: Orientation = ImageOrientation.Rotate180; break;
+				case 4: Orientation = ImageOrientation.FlipVertical; break;
+				case 5: Orientation = ImageOrientation.FlipVertical | ImageOrientation.Rotate270; break;
+				case 6: Orientation = ImageOrientation.Rotate90; break;
+				case 7: Orientation = ImageOrientation.FlipVertical | ImageOrientation.Rotate90; break;
+				case 8: Orientation = ImageOrientation.Rotate270; break;
+			}
+		}
+	}
+
+	[TargetSpecificImplementation, DotNetType("System.Drawing.Image")]
+	extern(CIL) class Image
+	{
+		[TargetSpecificImplementation]
+		public extern static Image FromStream(Stream stream);
+
+		[TargetSpecificImplementation]
+		public extern PropertyItem[] PropertyItems { get; }
+	}
+
+	[TargetSpecificImplementation, DotNetType("System.Drawing.Imaging.PropertyItem")]
+	extern(CIL) class PropertyItem
+	{
+		[TargetSpecificImplementation]
+		public extern int Id { get; set; }
+
+		[TargetSpecificImplementation]
+		public extern int Len { get; set; }
+
+		[TargetSpecificImplementation]
+		public extern short Type { get; set; }
+
+		[TargetSpecificImplementation]
+		public extern byte[] Value { get; set; }
+	}
+
+	extern(CIL) static class ExifDotNetImpl
+	{
+		const int OrientationTagId = 274; //as defined by the exif spec www.exif.org/Exif2-2.PDF (page 16)
+
+		internal static ExifData FromByteArray(byte[] bytes)
+		{
+			var memStream = new MemoryStream(bytes);
+			var img = Image.FromStream(memStream);
+			return new ExifData(GetOrientation(img));
```

Comment:
I guess `return new ExifData(GetOrientation(Image.FromStream(new MemoryStream(bytes))));` got a bit too hard to read? (I lost track of the parens myself while writing it :P)

How about 
```
var image = Image.FromStream(new MemoryStream(bytes));
return new ExifData(GetOrientation(image));
```